src/vtt-segment-loader.js

```diff
@@ -1006,44 +972,166 @@ export default class SegmentLoader extends videojs.EventTarget {
     let segmentInfo = this.pendingSegment_;
     let segment = segmentInfo.segment;
 
-    this.syncController_.probeSegmentInfo(segmentInfo);
+    // Make sure that vttjs has loaded, otherwise, wait till it finished loading
+    if (typeof window.WebVTT !== 'function' &&
+        this.subtitlesTrack_ &&
+        this.subtitlesTrack_.tech_) {
+
+      const loadHandler = () => {
+        this.subtitlesTrack_.tech_.off('vttjsloaded', loadHandler);
+        this.handleSegment_();
+      };
+
+      this.state = 'WAITING_ON_VTTJS';
+      this.subtitlesTrack_.tech_.on('vttjsloaded', loadHandler);
+      this.subtitlesTrack_.tech_.on('vttjserror', () => {
+        this.subtitlesTrack_.tech_.off('vttjsloaded', loadHandler);
+        this.error({
+          message: 'Error loading vtt.js'
+        });
+        this.state = 'READY';
+        this.pause();
+        this.trigger('error');
+      });
 
-    if (segmentInfo.isSyncRequest) {
-      this.trigger('syncinfoupdate');
-      this.pendingSegment_ = null;
-      this.state = 'READY';
       return;
     }
 
-    if (segmentInfo.timestampOffset !== null &&
-        segmentInfo.timestampOffset !== this.sourceUpdater_.timestampOffset()) {
-      this.sourceUpdater_.timestampOffset(segmentInfo.timestampOffset);
-    }
+    segment.requested = true;
 
-    // if the media initialization segment is changing, append it
-    // before the content segment
     if (segment.map) {
-      let initId = initSegmentId(segment.map);
+      // append WebVTT line terminators to the media initialization segment if it exists
+      // to follow the WebVTT spec (https://w3c.github.io/webvtt/#file-structure) that
+      // requires two or more WebVTT line terminators between the WebVTT header and the rest
+      // of the file
+      const initId = initSegmentId(segment.map);
+      const initSegment = this.initSegments_[initId];
+      const vttLineTerminators = new Uint8Array('\n\n'.split('').map(char => char.charCodeAt(0)));
+      const combinedByteLength = vttLineTerminators.byteLength + initSegment.bytes.byteLength;
+      const combinedSegment = new Uint8Array(combinedByteLength);
+
+      combinedSegment.set(initSegment.bytes);
+      combinedSegment.set(vttLineTerminators, initSegment.bytes.byteLength);
+      segment.map.bytes = combinedSegment;
```

Comment:
from #1043
@imbcmdth: Couple of thoughts on this whole block:

1. We should pre-calculate the `vttLineTerminators` value because it's a typed array that never changes.
2. I think we should do the entire append process when we first get the response for the `initSegment` so that we the object stored in `this.initSegments_[initId]` already has the value that would be in `combinedSegment`.